### PR TITLE
Error message for simulate evoked.

### PR DESCRIPTION
**Pull request by jaeilepp**
This is for testing the circle caching, but I thought I'd tackle an issue in the meantime.
Closes https://github.com/mne-tools/mne-python/issues/3370
See if this is good enough.

### CODE REVIEW

**Comment by jaeilepp** [comment]
Looks like the caching somewhat working. For PRs it is using the cache of the fork of the user. For this PR it only had a subset of the datasets because I had been playing around with the cache on my other branch. This could actually be the root of our problems somehow.

File: mne/simulation/evoked.py [hunk 1]
```diff
@@ -69,8 +69,14 @@ def simulate_evoked(fwd, stc, info, cov, snr=3., tmin=None, tmax=None,
     evoked = apply_forward(fwd, stc, info)
     if snr < np.inf:
         noise = simulate_noise_evoked(evoked, cov, iir_filter, random_state)
-        evoked_noise = add_noise_evoked(evoked, noise, snr,
-                                        tmin=tmin, tmax=tmax)
+        if evoked.data.shape != noise.data.shape:
```
Comment (Eric89GXL) [inline mne/simulation/evoked.py]: This check should probably be in `add_noise_evoked` somewhere, and also it should probably check that the set of channels is the same, not just that the data sizes are the same

Comment (Eric89GXL) [inline mne/simulation/evoked.py]: ... or actually I guess the error should be in `simulate_noise_evoked`

Comment (jaeilepp) [inline mne/simulation/evoked.py]: In ``simulate_noise_evoked`` we essentially just do
```
noise = evoked.copy()
noise.data = noise_data
```
, so the channels are actually the same anyway. Maybe this needs a more thorough look.

**Comment by jaeilepp** [comment]
Comment addressed.

**Comment by Eric89GXL** [comment]
LGTM can you add a test?

**Comment by jaeilepp** [comment]
> can you add a test?

Done.

**Comment by agramfort** [comment]
thx @jaeilepp

**Pull request merged**